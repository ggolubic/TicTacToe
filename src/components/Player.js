import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Player = ({ num, score }) => {
  return (
    <ScoreH2 num={num}>
      Player {num}: {score}
    </ScoreH2>
  );
};

Player.defaultProps = {
  num: 0,
  score: 0
};

Player.propTypes = {
  num: PropTypes.number.isRequired,
  score: PropTypes.number.isRequired
};

const ScoreH2 = styled.h2`
  color: ${props => (props.num === 1 ? "#545454" : "#ffd02d")};
`;

export default Player;
