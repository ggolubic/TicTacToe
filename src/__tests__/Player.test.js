import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import "jest-styled-components";

import Player from "../components/Player";

Enzyme.configure({ adapter: new Adapter() });

describe("Player component", () => {
  it("Should render correctly with default props", () => {
    const tree = renderer.create(<Player />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render correctly with passed props", () => {
    const props = setup();
    const tree = renderer.create(<Player {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should display correct color depending on player", () => {
    const { score } = setup();
    const player1 = mount(<Player num={1} score={score} />);
    expect(player1).toHaveStyleRule("color", "#545454");
    player1.unmount();
    const player2 = mount(<Player num={2} score={score} />);
    expect(player2).toHaveStyleRule("color", "#ffd02d");
    player2.unmount();
  });
});

function setup() {
  const props = {
    num: 1,
    score: 0
  };
  return { ...props };
}
