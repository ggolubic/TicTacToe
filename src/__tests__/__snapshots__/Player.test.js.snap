// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component Should render correctly with default props 1`] = `
.c0 {
  color: #ffd02d;
}

<h2
  className="c0"
>
  Player 
  0
  : 
  0
</h2>
`;

exports[`Player component Should render correctly with passed props 1`] = `
.c0 {
  color: #545454;
}

<h2
  className="c0"
>
  Player 
  1
  : 
  0
</h2>
`;
